test/e2e/bootstrapper_test.go

```diff
@@ -101,7 +103,12 @@ func TestBootstrapper(t *testing.T) {
 	// Does not include node-labels and container-image since their paths do not depend on underlying OS
 	checkPathsFor := []string{"--bootstrap-kubeconfig", "--cloud-config", "--config", "--kubeconfig", "--log-file",
 		"--cert-dir"}
-	expectedDependencies := []string{"docker"}
```

Comment:
You can keep docker as default in `expectedDependencies` and overwrite if `wmcb.dockerRuntime` evaluates to false.